proto/raft.proto

```diff
@@ -706,3 +706,20 @@ message GetRangeDebugInfoResponse {
   bool has_lease = 4;
   RaftMembership membership = 5;
 }
+
+message RangeDescriptorChange {
+  RangeDescriptor old_descriptor = 1;
+  RangeDescriptor new_descriptor = 2;
+}
+
+message PartitionDescriptorChange {
+  PartitionDescriptor old_descriptor = 1;
+  PartitionDescriptor new_descriptor = 2;
+}
+
+message AdminUpdateDescriptorRequest {
+	RangeDescriptorChange range_descriptor_change = 1;
+	repeated PartitionDescriptorChange partition_descriptor_change = 2;
```

Comment:
I was a little surprised the PartitionDescriptorChange was repeated but the RangeDescriptorChange was not -- why is it like that?

Reply:
we need a distributed transaction to update range descriptor, so it's not possible to batch the range descriptor updates. However, we can batch the partition descriptor updates.